examples/tv-casting-app/darwin/MatterTvCastingBridge/MatterTvCastingBridge/CastingServerBridge.mm

```diff
@@ -120,10 +111,166 @@ - (instancetype)init
     return self;
 }
 
+/**
+ Dispatches the block on the Matter SDK dispatch queue.
+
+ @param description Optional. A description of the callback to be dispatched for diagnostic purposes.
+ @param block The Block to be dispatched.
+ */
+- (void)dispatchOnMatterSDKQueue:(const NSString * _Nullable)description block:(dispatch_block_t)block
+{
+    if (nil != description) {
+        ChipLogProgress(AppServer, "[SYNC] CastingServerBridge %s", [description UTF8String]);
+    }
+    dispatch_async(_chipWorkQueue, ^{
+        // The Matter SDK is not, generally, internally thread-safe. It relies upon the client to lock
+        // the SDK prior to any interactions after chip::DeviceLayer::PlatformMgr().InitChipStack()
+        //
+        // Note that it is presently safe to do at this point because InitChipStack is called in the
+        // constructor for the CastingServerBridge.
+        chip::DeviceLayer::StackLock lock;
```

Comment:
If StackLock were _not_ a no-op on Darwin, this would deadlock, since you are already on the Matter queue and hence holding the lock if there were one.